AmazonChimeSDKDemo/AmazonChimeSDKDemo/LiveTranscriptionOptionsViewController.swift

```
@@ -139,31 +230,153 @@ class LiveTranscriptionOptionsViewController: UIViewController, UITextFieldDeleg
+                piiEntityTypes = piiContentRedactionSelected
+                contentRedactionType = transcribeContentIdetificationType
+            }
+        }
+        
+        if partialResultsStabilizationOption > 0 {
+            enablePartialResultsStabilization = true
+            if partialResultsStabilizationOption == 1 {
+                partialResultsStabilty = partialResultsStabilizationsList.last?.lowercased()
+            } else {
+                partialResultsStabilty = partialResultsStabilizationsList[partialResultsStabilizationOption].lowercased()
+            }
+        }
+        
+        if enableCustomLangugeModelTextFieldSwitch.isOn {
+            if (customLanguageModelTextField.text?.isEmpty) == true {
+                self.view.makeToast("Custom language model name cannot be empty!")
+                return
+            } else {
+                languageModelName = customLanguageModelTextField.text
+            }
+        }
+        
+        if enableAutomaticLanguageIdentificationSwitch.isOn {
+            if !preferredLanguageOptions.isEmpty {
+                preferredLanguage = preferredLanguageOptions
+            }
+            languageOptions = selectedLanguageOptions
+            identifyLanguage = true
+            languageSelected = ""
+        }
+        
+        let transcriptionStreamParams = TranscriptionStreamParams(
+            contentIdentificationType: contentIdentificationType,
+            contentRedactionType: contentRedactionType,
+            enablePartialResultsStabilization: enablePartialResultsStabilization,
+            partialResultsStability: partialResultsStabilty,
+            piiEntityTypes: piiEntityTypes,
+            languageModelName: languageModelName,
+            identifyLanguage: identifyLanguage,
+            languageOptions: languageOptions,
+            preferredLanguage: preferredLanguage)
+        
+        let encodedData = try? JSONEncoder().encode(transcriptionStreamParams)
```

this will fail silently right? should we not handle the catch explicitly?

---

it will try to encode the data, if it fails will give null. Transcription will still start  if `transcriptionStreamParams` is null.